### PR TITLE
[tech] Skeleton for the Netex France exporter

### PR DESCRIPTION
**Pull request by woshilapin**
Just a skeleton for the exporter. Trying something new in term of architecture with a `struct` (instead of a function) that will own the `Model` and be responsible for the export. The reason is that we discussed a lot of possible parameters to pass to the binary; the `Exporter` is the structure that would also have these parameters.

Note: Putting that PR as a Draft for now. Please tell me what you think of it.

### CODE REVIEW

File: src/netex_france/exporter.rs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+{
+    /// Build a Netex France profile exporter from the model.
+    /// `path` is the expected output Path where the Netex France is going to be
+    /// written. It should be a folder that already exists.
+    pub fn new(model: Model, path: P) -> Self {
+        Exporter { model, path }
+    }
+
+    /// Actually write `model` into `path` as a Netex France profile.
+    pub fn write(&self) -> Result<()> {
```
Comment (datanel) [inline src/netex_france/exporter.rs]: I prefer `write(&self, path: P)` if path is not used elsewhere

Comment (woshilapin) [reply src/netex_france/exporter.rs]: Good point!

**Comment by woshilapin** [comment]
In last commit 742b1a4afd4313f5451a96ab2283634f59499087, I added a few things:
- A new parameter to `Exporter` object which will contain the `ParticipantRef` value (an input parameter of the program)
- A method `write_stops` which returns an `Element` which should be a `GeneralFrame`
- A method to wrap the frame (see above) inside a entire NeTEx block.

File: src/netex_france/exporter.rs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+use chrono::prelude::*;
+use minidom::Element;
+use std::{
+    convert::AsRef,
+    fmt::{self, Display, Formatter},
+    fs::File,
+    path::Path,
+};
+
+const NETEX_FRANCE_STOPS_FILENAME: &'static str = "arrets.xml";
```
Comment (datanel) [inline src/netex_france/exporter.rs]: clippy -> `&str`

File: src/netex_france/exporter.rs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+            .ns("http://www.netex.org.uk/netex/")
+            .append(frame)
+            .build();
+        let root = Element::builder("PublicationDelivery")
+            .ns("http://www.netex.org.uk/netex/")
+            .attr("xmlns:xsi", "http://www.w3.org/2001/XMLSchema-instance")
+            .attr("version", format!("1.09:FR-NETEX_{}-2.1-1.0", version_type))
+            .append(publication_timestamp)
+            .append(participant_ref)
+            .append(data_objects)
```
Comment (datanel) [inline src/netex_france/exporter.rs]: I'm not sure but this node is not wrapped by `<CompositeFrame><frames></frames></CompositeFrame>`?

Comment (woshilapin) [reply src/netex_france/exporter.rs]: This is still something to discuss with @prhod. From what I understood of NeTEx specification (well, the XSD files), you need to wrap in a `CompositeFrame` only if there is more than one frame. In the fixtures we do have for IDFM, they have 2 `GeneralFrame` but only one contains the stops. The other one seems useless and we might not produce it for NeTEx France. So if we do have only one to produce, we might just skip this extra level of `CompositeFrame`. But as I said, still to be discussed with @prhod.

File: src/netex_france/exporter.rs (hunk 2 above)
Comment (woshilapin) [reply src/netex_france/exporter.rs]: Yep, I just pushed that after I saw the CI complaining :smile:

**Comment by woshilapin** [comment]
One more thing, it'd be nice that you also update the specifications as part of this PR.

File: src/netex_france/exporter.rs (hunk 3 above)
Comment (woshilapin) [reply src/netex_france/exporter.rs]: So, decision has been taken to go for the simplest solution for now (without a `CompositeFrame`). We're not 100% sure it's completely correct yet but it should fairly easy to wrap it in a `CompositeFrame` later if needed.

**Comment by woshilapin** [comment]
I just added this commit 855e57cb6fb6efd0af1baa5d5b8809473190a2f9 to change the way were gonna write timestamp in the file.

**Comment by datanel** [comment]
@Mergifyio refresh

**Comment by ArnaudOggy** [comment]
@Mergifyio refresh

**Pull request merged**